test/e2e/app-dir/app-external/app-external.test.ts

```diff
@@ -251,21 +251,37 @@ createNextDescribe(
       expect(html).toContain('success')
     })
 
-    it('should not prefer to resolve esm over cjs for bundling optout packages', async () => {
-      const browser = await next.browser('/optout/action')
-      expect(await browser.elementByCss('#dual-pkg-outout p').text()).toBe('')
-
-      browser.elementByCss('#dual-pkg-outout button').click()
-      await check(async () => {
-        const text = await browser.elementByCss('#dual-pkg-outout p').text()
-        if (process.env.TURBOPACK) {
-          // The prefer esm won't effect turbopack resolving
-          expect(text).toBe('dual-pkg-optout:mjs')
-        } else {
-          expect(text).toBe('dual-pkg-optout:cjs')
-        }
-        return 'success'
-      }, /success/)
+    describe('server actions', () => {
+      it('should not prefer to resolve esm over cjs for bundling optout packages', async () => {
+        const browser = await next.browser('/optout/action')
+        expect(await browser.elementByCss('#dual-pkg-outout p').text()).toBe('')
+
+        browser.elementByCss('#dual-pkg-outout button').click()
+        await check(async () => {
+          const text = await browser.elementByCss('#dual-pkg-outout p').text()
+          if (process.env.TURBOPACK) {
+            // The prefer esm won't effect turbopack resolving
+            expect(text).toBe('dual-pkg-optout:mjs')
+          } else {
+            expect(text).toBe('dual-pkg-optout:cjs')
+          }
+          return 'success'
+        }, /success/)
+      })
+
+      if (!process.env.TURBOPACK) {
```

Comment:
Avoid that. Just keep it unconditional and use the turbopack-test-manifest to flag it as failed. This way we can track the progress of that.